intake/catalog/tests/test_local.py

```diff
@@ -75,7 +75,7 @@ def test_source_plugin_config(catalog1):
 
 def test_metadata(catalog1):
     assert hasattr(catalog1, 'metadata')
-    assert catalog1['metadata']['test'] is True
+    assert catalog1.metadata['test'] is True
```

Comment:
I feel like the change may break things for too many people. I note that for a *local* catalog, both methods still work. Also, it is inkeeping with the idea that `in` normally relates to what you can do with getitem. This is inspired by the usage of both in Pandas.